engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/CloudOrchestrator.java

```diff
@@ -200,48 +204,50 @@ public VirtualMachineEntity createVirtualMachine(String id, String owner, String
             }
         }
 
-        if (vm.getDiskOfferingId() != null) {
-            DiskOfferingVO diskOffering = _diskOfferingDao.findById(vm.getDiskOfferingId());
+        if (diskOfferingId != null) {
+            DiskOfferingVO diskOffering = _diskOfferingDao.findById(diskOfferingId);
             if (diskOffering == null) {
-                throw new InvalidParameterValueException("Unable to find disk offering " + vm.getDiskOfferingId());
+                throw new InvalidParameterValueException("Unable to find disk offering " + diskOfferingId);
```

Comment:
```suggestion
                throw new InvalidParameterValueException("Unable to find data disk offering " + dataDiskOfferingId);
```

Update to 'data' disk here if relevant.